cliclient/cliclient.go

```diff
@@ -25,6 +29,29 @@ type MasterClient struct {
 	CAPIClient       *capiClient.Client
 }
 
+// HTTPClienter is a http.Client factory interface
+type HTTPClienter interface {
+	New() *http.Client
+}
+
+// DefaultHTTPClient stores the default http.Client factory
+var DefaultHTTPClient HTTPClienter = &HTTPClient{}
+
+/*
+TestingReplaceDefaultHTTPClient replaces DefaultHTTPClient for unit tests.
+Not thread-safe.
+Call the returned function by defer keyword, for example:
+
+	defer cliclient.TestingReplaceDefaultHTTPClient(mockClient)()
+*/
+func TestingReplaceDefaultHTTPClient(mockClient HTTPClienter) func() {
```

Comment:
Can we move it undert the `cliclient_test.go` file?

Reply:
Yes, we can. But the compile will be failed, because this function is called from another package, too (`cmd/*_test.go`). Go compiler enables importing from `*_test.go files` only from same package.

It's possible to create a new package, for example `testutil`, which can be imported from everywhere. But it may introduce a new claim: a new package. But anyway, if you mind, I can do it.

Reply:
I think that creating a new package would be a better approach. What do you think @crobby ?